src/DFMM.sol

```diff
@@ -259,16 +260,19 @@ contract DFMM is IDFMM {
         state.tokenIn = _pools[poolId].tokens[state.tokenInIndex];
         state.tokenOut = _pools[poolId].tokens[state.tokenOutIndex];
 
-        address[] memory tokens = new address[](1);
-        tokens[0] = state.tokenIn;
-        uint256[] memory amounts = new uint256[](1);
-        amounts[0] = state.amountIn;
-
         // Optimistically transfer the output tokens to the recipient.
         _transfer(state.tokenOut, recipient, state.amountOut);
 
+        IStrategy(_pools[poolId].strategy).postSwapHook(
+            msg.sender, poolId, _pools[poolId], state.postSwapHookData
+        );
```

Comment:
Is it possible to make this a "pre-swap" hook? Could you describe the details of this in the pr more too for reference

Reply:
it's post swap because you need to compute the appropriate swap data off chain and once it's validated that becomes the next state, there's nothing you can do preswap here unless you bisect over the nextL which is kind of a nonstarter here imo